src/topology/sheaves/epi_mono.lean

```diff
@@ -0,0 +1,97 @@
+/-
+Copyright (c) 2022 Jujian Zhang. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Jujian Zhang
+-/
+import topology.sheaves.sheaf
+import category_theory.sites.sheafification
+
+/-!
+# Epimorphisms and monomorphisms in category of (pre)sheaves
+
+Let `X` be a topological space and `F, G` be sheaves on `X`. Then `f : F ⟶ G` is monic iff it is
+monic as presheaf morphisms; `f` is epic if `f` is epic as presheaf morphism, but in this case, the
+converse is **not** necessarily true.
+
+-/
+
+
+universes v u
+
+open category_theory category_theory.limits
+open topological_space
+open Top
+
+
+variables {X : Top.{v}} {D : Type u} [category.{v u} D]
+
+section sheaf
+
+variables {F G : sheaf D X} (f : F ⟶ G)
+
+-- Not an instance to prevent class search forming a loop
+lemma mono_of_presheaf_mono [mono f.1] : mono f :=
+{ right_cancellation := λ P g h eq₀, Sheaf.hom.ext _ _ $ (cancel_mono f.1).mp $
+    (Sheaf.hom.ext_iff _ _).mp eq₀ }
+
+instance sheaf.forget_reflects_mono : (sheaf.forget D X).reflects_monomorphisms :=
+{ reflects := λ F G f m, @@mono_of_presheaf_mono _ f m }
```

Comment:
```suggestion
instance sheaf.forget_reflects_mono : (sheaf.forget D X).reflects_monomorphisms :=
infer_instance
```
Since faithful functors always reflects monos and epis